float64 reward
bool done
---
bool ret_value